crates/solidity/outputs/cargo/tests/src/doc_examples/cursor_api.rs

```diff
@@ -0,0 +1,80 @@
+use anyhow::Result;
+use semver::Version;
+
+use slang_solidity::{
+    language::Language,
+    syntax::{
+        nodes::{RuleKind, TokenKind},
+        parser::ProductionKind,
+    },
+};
+
+#[test]
```

Comment:
I suggest splitting into three tests for readability.

Reply:
I thought about that, but because the 'test' nature of this is not actually for testing, but just to validate that the example is ok, I decided that the clarity I could gain by simply repeating a section of code with exactly the same result but using different API patterns was worth having the repeated functionality.

Reply:
I think it is also for "testing", since the Rust test infra is the same in all projects, this is a quick way for someone to just copy a test locally, and edit/run it for experimenting.

It also indicates to readers that these are separate, instead of having side effects on later statements.